configs/tsconfig.json

```diff
@@ -1,9 +1,6 @@
 {
   "compilerOptions": {
```

Comment:
Does all the tooling know to look for config here?

Reply:
Do we need 3 configs? Can we make the esm one the base?

Reply:
Also you're right about tooling, I moved the tsconfigs to the root. I made the cjs one the base because mocha does not play well when module is set to "es2022" in the tsconfig, and because I think it makes sense that the base tsconfig is the same as what we had previously.